tests/php/Tasks/InvalidateAllSessionsTaskTest.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace SilverStripe\SessionManager\Tests\Services;
+
+use SilverStripe\Control\Middleware\ConfirmationMiddleware\Url;
+use SilverStripe\Control\Tests\HttpRequestMockBuilder;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\Dev\SapphireTest;
+use SilverStripe\ORM\FieldType\DBDatetime;
+use SilverStripe\Security\AuthenticationHandler;
+use SilverStripe\Security\Member;
+use SilverStripe\Security\RememberLoginHash;
+use SilverStripe\Security\Security;
+use SilverStripe\SessionManager\Extensions\RememberLoginHashExtension;
+use SilverStripe\SessionManager\Middleware\LoginSessionMiddleware;
+use SilverStripe\SessionManager\Models\LoginSession;
+use SilverStripe\SessionManager\Tasks\InvalidateAllSessionsTask;
+
+class InvalidateAllSessionsTaskTest extends SapphireTest
+{
+    use HttpRequestMockBuilder;
+
+    protected static $fixture_file = 'InvalidateAllSessionsTaskTest.yml';
+
+    protected static $required_extensions = [
+        RememberLoginHash::class => [
+            RememberLoginHashExtension::class,
+        ],
+    ];
+
+    public function testRemoveAllSessions()
+    {
+        // Ensure existing fixtured sessions are valid
+        DBDatetime::set_mock_now('2003-02-15 10:00:00');
+
+        // Log in as member 1 and make them active
+        $request = $this->buildRequestMock('dev/tasks/' . InvalidateAllSessionsTask::config()->get('segment'));
+        $request->method('getIP')->willReturn('192.168.0.1');
+        $member = $this->objFromFixture(Member::class, 'member1');
+        $handler = Injector::inst()->get(AuthenticationHandler::class);
+        $handler->login($member, true, $request);
+
+        // Ensure fixtured sessions are created and at least one login hash is generated
+        $this->assertNotSame(0, LoginSession::get()->count(), 'There are no LogginSessions');
+        $this->assertNotSame(0, RememberLoginHash::get()->count(), 'There are no RememberLoginHashes');
+
+        // Completely invalidate all current sessions and login hashes
+        $task = new InvalidateAllSessionsTask();
+        $task->run($request);
+
+        // Assert all sessions are removed from the database
+        $this->assertSame(0, LoginSession::get()->count(), 'There is at least one LoginSession');
+        $this->assertSame(0, RememberLoginHash::get()->count(), 'There is at least one RememberLoginHash');
+
+        // Navigate to re-validate current session
+        $middleware = new LoginSessionMiddleware(new Url('no-match'));
+        $next = false;
+        $middleware->process(
+            $request,
+            function () use (&$next) {
+                $next = true;
```

Comment:
```suggestion
                // noop
```
We don't need the $next var?

Reply:
Fair enough. I just copied this from a similar test elsewhere in the module, and I don't understand middleware deeply enough yet to fully know how `$next` is used. 😅